TruePath.Tests/PathExtensionsTests.cs

```diff
@@ -73,4 +73,52 @@ public void FileNameInvariantTests(string inputPath)
         var fileName = path.FileName;
         Assert.Equal(fileName, path.GetFilenameWithoutExtension() + path.GetExtensionWithDot());
     }
+
+    [Theory]
+    [InlineData(@"C:\", "bar", @"C:\.bar")]
+    [InlineData(@"C:\filename.foo", "bar", @"C:\filename.bar")]
+    [InlineData(@"\", "bar", @"\.bar")]
+    [InlineData(@"\file", "bar", @"\file.bar")]
+    [InlineData(@"\file", ".bar", @"\file.bar")]
+    [InlineData(@"\file.", ".bar", @"\file.bar")]
+    [InlineData("file.foo", "bar", "file.bar")]
+    public void WithExtensionTests_Windows(string inputPath, string newExtension, string expected)
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return;
+        }
+
+        // Arrange
+        var path = new LocalPath(inputPath);
+
+        // Act
+        var newPath = path.WithExtension(newExtension);
+
+        // Assert
+        Assert.Equal(expected, newPath.Value);
+    }
+
+    [Theory]
+    [InlineData("/", "bar", "/.bar")]
+    [InlineData("/file", "bar", "/file.bar")]
+    [InlineData("/file", ".bar", "/file.bar")]
+    [InlineData("/file.", ".bar", "/file.bar")]
+    [InlineData("file.foo", "bar", "file.bar")]
+    public void WithExtensionTests_Unix(string inputPath, string @newExtension, string expected)
```

Comment:
Let's add test cases (and clarify behavior) for cases like `.gitignore`.

(I believe this is the whole filename consisting of extension only? So updating it should replace the whole file name?)